csrc/multidevice/communication.h

```diff
@@ -48,21 +49,29 @@ class Communication : public Expr {
   // Only specify `root` for types that have root.
   // Only specify `red_op` for reduction types.
   // Only specify `scattered_axis` for ReduceScatter.
-  //
-  // TODO: pass in input/output TV and compute root, mesh and scatteredAxis from
-  // them.
   Communication(
       IrBuilderPasskey passkey,
       CommunicationType type,
-      DeviceMesh mesh, // Might not contain `root`.
+      TensorView* out,
+      TensorView* in,
       Team team, // All devices involved in this communication. It must include
                  // `root`. It can be a subset of `root`+`mesh` in case of 2D
                  // sharding.
       DeviceIdxType root = -1,
       RedOpType red_op = RedOpType::UNUSED,
-      int64_t scattered_axis = -1,
-      TensorView* input_tv = nullptr,
-      TensorView* output_tv = nullptr);
+      int64_t scattered_axis = -1);
+
+  // Currently, only used for MultiDeviceExecutor.
+  Communication(
+      IrBuilderPasskey passkey,
+      CommunicationType type,
+      DeviceMesh mesh,
+      Team team, // All devices involved in this communication. It must include
+                 // `root`. It can be a subset of `root`+`mesh` in case of 2D
+                 // sharding.
+      DeviceIdxType root = -1,
+      RedOpType red_op = RedOpType::UNUSED,
+      int64_t scattered_axis = -1);
```

Comment:
Could you comment when to use which version?

Reply:
Done -- updated the comment.